### PR TITLE
fix(js): skip tsc batch builds for implicit dependencies

### LINKED ISSUE #28839: @nx/js:tsc executor is required for implicitDependencies in batch mode
### Current Behavior

When I create a workspace with both a TypeScript and Python project, with the TypeScript project having an implicit dependency on the Python one, the TypeScript project fails to compile in batch mode with the error:

```
The "@nx/js:tsc" batch executor requires all dependencies to use the "@nx/js:tsc" executor.
None of the following tasks in the "my-python-project" project use the "@nx/js:tsc" executor:
- my-python-project:build
```

### Expected Behavior

The `@nx/js:tsc` executor should not attempt to incrementally build implicit dependencies, since this indicates there isn't an explicit reference in code, and therefore incremental builds with `tsc` aren't applicable.

### GitHub Repo

https://github.com/cogwirrel/nx-tsc-batch-implicit-deps-example

### Steps to Reproduce

1. Clone the [example repo](https://github.com/cogwirrel/nx-tsc-batch-implicit-deps-example)
2. `pnpm install`
3. `pnpm nx run-many --target build --batch --all --verbose`
4. Observe error

Alternatively set up from scratch:

1. `pnpm dlx create-nx-workspace --preset=ts --ci=skip`
2. `pnpm exec nx generate @nx/js:library --directory=packages/my-js-lib --bundler=tsc --linter=none --name=my-js-lib --unitTestRunner=none --no-interactive`
3. Add some other project (eg a python one)
4. Add an implicit dependency on the other project in `packages/my-js-lib/project.json`
5. Build in batch mode and observe error

### Nx Report

```
Node           : 20.18.0
OS             : darwin-arm64
Native Target  : aarch64-macos
pnpm           : 9.12.3

nx             : 20.0.11
@nx/js         : 20.0.11
@nx/workspace  : 20.0.11
@nx/devkit     : 20.0.11
@nrwl/devkit   : 19.8.10
@nrwl/tao      : 19.8.10
typescript     : 5.5.4
---------------------------------------
Registered Plugins:
@nx/js/typescript
---------------------------------------
Community plugins:
@nxlv/python : 19.2.1
---------------------------------------
The following packages should match the installed version of nx
  - @nrwl/devkit@19.8.10
  - @nrwl/tao@19.8.10

To fix this, run `nx migrate nx@20.0.11`
```

### Failure Logs

```
> nx run my-js-lib:build

The "@nx/js:tsc" batch executor requires all dependencies to use the "@nx/js:tsc" executor.
None of the following tasks in the "my-python-project" project use the "@nx/js:tsc" executor:
- my-python-project:build
```

### Package Manager Version

pnpm 9.12.3

### Operating System

- [x] macOS
- [ ] Linux
- [ ] Windows
- [ ] Other (Please specify)

### Additional Information

I believe it should work as expected if the following filters out tasks for projects which are implicit dependencies:

https://github.com/nrwl/nx/blob/6882ad14e401df4331f69d80780f0921894da66d/packages/js/src/executors/tsc/lib/get-tsconfig.ts#L101-L109

### PR DESCRIPTION
**Pull request by cogwirrel**
Implicit dependencies are not referenced in code and therefore TSC incremental builds are not applicable.

## Current Behavior
A project using the `@nx/js:tsc` executor will fail to build if it has implicit dependencies on projects which do not use the `@nx/js:tsc` executor.

To reproduce:
* Clone https://github.com/cogwirrel/nx-tsc-batch-implicit-deps-example
* `pnpm i && pnpm nx run-many --target build --batch --all`

## Expected Behavior
- Implicit dependencies that do not use the `@nx/js:tsc` executor are permitted. For example, a TypeScript project may implicitly depend on a Python project, but the TypeScript project should still be buildable in batch mode.
- Projects using the `@nx/js:tsc` executor will still fail to build if they have explicit dependencies on projects which do not use the `@nx/js:tsc` executor.

Tested by publishing to the local registry, upgrading the [example repo](https://github.com/cogwirrel/nx-tsc-batch-implicit-deps-example) to use my local version, and built successfully in batch mode.

## Related Issue(s)
Fixes #28839

### CODE REVIEW

File: packages/js/docs/tsc-examples.md [hunk 1]
```diff
@@ -85,7 +85,7 @@ Executing tasks in batch mode is an experimental feature.
 {% /callout %}
 
 {% callout type="info" title="Requirements" %}
-Building a project with the `@nx/js:tsc` executor in batch mode requires all dependent projects to be buildable and built using the `@nx/js:tsc` executor.
+Building a project with the `@nx/js:tsc` executor in batch mode requires all dependent projects (besides implicit dependencies) to be buildable and built using the `@nx/js:tsc` executor.
```
Comment (leosvelperez) [inline packages/js/docs/tsc-examples.md:88]:
```suggestion
Building a project with the `@nx/js:tsc` executor in batch mode requires all dependent projects (excluding implicit dependencies) to be buildable and built using the `@nx/js:tsc` executor.
```

**Review by leosvelperez — CHANGES_REQUESTED** [review]
Thanks for the contribution!

It looks good to me in general. I just left a comment with a grammar change. Please remember to run `pnpm documentation` to re-generate the docs.

**Comment by cogwirrel** [comment]
> Thanks for the contribution!
> 
> It looks good to me in general. I just left a comment with a grammar change. Please remember to run `pnpm documentation` to re-generate the docs.

Thanks so much for the review! I've updated the wording in the docs :)

**Review by leosvelperez — APPROVED** [review]
Thank you!

**Pull request merged**